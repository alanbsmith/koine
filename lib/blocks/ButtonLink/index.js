import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

import Button from '../Button';

import gridScale from '../../utils/gridScale';

const modifierConfig = {
  hoverCustom: ({ theme, hoverColor }) => `
    &:hover, &:focus {
      color: ${theme.colors.base[hoverColor]};
    }
  `,

  hoverDanger: ({ theme }) => `
    &:hover, &:focus {
      color: ${theme.colors.status.danger};
    }
  `,

  hoverInfo: ({ theme }) => `
    &:hover, &:focus {
      color: ${theme.colors.status.info};
    }
  `,

  hoverSuccess: ({ theme }) => `
    &:hover, &:focus {
      color: ${theme.colors.status.success};
    }
  `,

  hoverWarning: ({ theme }) => `
    &:hover, &:focus {
      color: ${theme.colors.status.warning};
    }
  `,

  fullWidth: () => `flex: 1`,
};

const ButtonLink = styled(Button)`
  background: transparent;
  color: ${props => props.theme.colors.base.chrome600};
  font-size: ${props => gridScale(props, 1.5)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;

  &:hover,
  &:focus {
    background: transparent;
    box-shadow: none;
    color: ${props => props.theme.colors.base.chrome900};
    text-decoration: underline;
  }

  ${applyStyleModifiers(modifierConfig)}
`;

ButtonLink.defaultProps = {
  hoverColor: 'chrome900',
};

export default ButtonLink;
