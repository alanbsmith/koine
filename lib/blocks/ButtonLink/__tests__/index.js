import React from 'react';
import ButtonLink from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('ButtonLink Block', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <ButtonLink>
        Login
      </ButtonLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the bgDanger modifier', () => {
    const tree = renderWithTheme(
      <ButtonLink modifiers={['bgDanger']}>
        Login
      </ButtonLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverBgDanger modifier', () => {
    const tree = renderWithTheme(
      <ButtonLink modifiers={['hoverBgDanger']}>
        Login
      </ButtonLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the fullWidth modifier', () => {
    const tree = renderWithTheme(
      <ButtonLink modifiers={['fullWidth']}>
        Login
      </ButtonLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverCustom modifier', () => {
    const tree = renderWithTheme(
      <ButtonLink
        hoverColor="pink"
        modifiers={['hoverCustom']}
      >
        Login
      </ButtonLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('uses default styles if no hoverColor is provided with hoverCustom modifier', () => {
    const tree = renderWithTheme(
      <ButtonLink
        modifiers={['hoverCustom']}
      >
        Login
      </ButtonLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', '#1F2D3D', {
      modifier: ':hover',
    });
  });

  it('adds the hoverDanger modifier', () => {
    const tree = renderWithTheme(
      <ButtonLink modifiers={['hoverDanger']}>
        Login
      </ButtonLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverInfo modifier', () => {
    const tree = renderWithTheme(
      <ButtonLink modifiers={['hoverInfo']}>
        Login
      </ButtonLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverSuccess modifier', () => {
    const tree = renderWithTheme(
      <ButtonLink modifiers={['hoverSuccess']}>
        Login
      </ButtonLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverWarning modifier', () => {
    const tree = renderWithTheme(
      <ButtonLink modifiers={['hoverWarning']}>
        Login
      </ButtonLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
