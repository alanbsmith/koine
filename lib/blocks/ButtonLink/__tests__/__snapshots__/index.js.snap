// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink Block adds the bgDanger modifier 1`] = `
.c0 {
  background: transparent;
  color: #646464;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus {
  background: transparent;
  box-shadow: none;
  color: #191919;
  text-decoration: underline;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  background-color: #FF4949;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

<button
  className="c0 c1"
>
  Login
</button>
`;

exports[`ButtonLink Block adds the fullWidth modifier 1`] = `
.c0 {
  background: transparent;
  color: #646464;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0:hover,
.c0:focus {
  background: transparent;
  box-shadow: none;
  color: #191919;
  text-decoration: underline;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

<button
  className="c0 c1"
>
  Login
</button>
`;

exports[`ButtonLink Block adds the hoverBgDanger modifier 1`] = `
.c0 {
  background: transparent;
  color: #646464;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus {
  background: transparent;
  box-shadow: none;
  color: #191919;
  text-decoration: underline;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c1:hover,
.c1:focus {
  background-color: #ff6262;
}

<button
  className="c0 c1"
>
  Login
</button>
`;

exports[`ButtonLink Block adds the hoverCustom modifier 1`] = `
.c0 {
  background: transparent;
  color: #646464;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus {
  background: transparent;
  box-shadow: none;
  color: #191919;
  text-decoration: underline;
}

.c0:hover,
.c0:focus {
  color: undefined;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

<button
  className="c0 c1"
>
  Login
</button>
`;

exports[`ButtonLink Block adds the hoverDanger modifier 1`] = `
.c0 {
  background: transparent;
  color: #646464;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus {
  background: transparent;
  box-shadow: none;
  color: #191919;
  text-decoration: underline;
}

.c0:hover,
.c0:focus {
  color: #FF4949;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

<button
  className="c0 c1"
>
  Login
</button>
`;

exports[`ButtonLink Block adds the hoverInfo modifier 1`] = `
.c0 {
  background: transparent;
  color: #646464;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus {
  background: transparent;
  box-shadow: none;
  color: #191919;
  text-decoration: underline;
}

.c0:hover,
.c0:focus {
  color: #1FB6FF;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

<button
  className="c0 c1"
>
  Login
</button>
`;

exports[`ButtonLink Block adds the hoverSuccess modifier 1`] = `
.c0 {
  background: transparent;
  color: #646464;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus {
  background: transparent;
  box-shadow: none;
  color: #191919;
  text-decoration: underline;
}

.c0:hover,
.c0:focus {
  color: #13CE66;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

<button
  className="c0 c1"
>
  Login
</button>
`;

exports[`ButtonLink Block adds the hoverWarning modifier 1`] = `
.c0 {
  background: transparent;
  color: #646464;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus {
  background: transparent;
  box-shadow: none;
  color: #191919;
  text-decoration: underline;
}

.c0:hover,
.c0:focus {
  color: #FFC82C;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

<button
  className="c0 c1"
>
  Login
</button>
`;

exports[`ButtonLink Block renders correctly 1`] = `
.c0 {
  background: transparent;
  color: #646464;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus {
  background: transparent;
  box-shadow: none;
  color: #191919;
  text-decoration: underline;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

<button
  className="c0 c1"
>
  Login
</button>
`;

exports[`ButtonLink Block uses default styles if no hoverColor is provided with hoverCustom modifier 1`] = `
.c0 {
  background: transparent;
  color: #646464;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus {
  background: transparent;
  box-shadow: none;
  color: #191919;
  text-decoration: underline;
}

.c0:hover,
.c0:focus {
  color: #191919;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

<button
  className="c0 c1"
>
  Login
</button>
`;
